rskj-core/src/test/java/co/rsk/peg/BridgeTest.java

```diff
@@ -2298,6 +2301,136 @@ public void mineableMethodsDefinition() {
+            }
+            Assert.assertEquals(cost, bridge.getGasForData(data));
+        }
+    }
+
+    @Test
+    public void receiveHeadersAccess_beforePublic_noAccessIfNotFromFederationMember() throws Exception {
+        doReturn(false).when(activationConfig).isActive(eq(RSKIP124), anyLong());
+
+        RskAddress sender = new RskAddress("2acc95758f8b5f583470ba265eb685a8f45fc9d5");
+        Transaction txMock = mock(Transaction.class);
+        when(txMock.getReceiveAddress()).thenReturn(PrecompiledContracts.BRIDGE_ADDR);
+        when(txMock.getSender()).thenReturn(sender);
+
+        BridgeSupport bridgeSupportMock = mock(BridgeSupport.class);
+        when(bridgeSupportMock.getRetiringFederation()).thenReturn(null);
+        when(bridgeSupportMock.getActiveFederation()).thenReturn(BridgeRegTestConstants.getInstance().getGenesisFederation());
+
+        Bridge bridge = PowerMockito.spy(new Bridge(PrecompiledContracts.BRIDGE_ADDR, constants, activationConfig));
+        PowerMockito.doReturn(bridgeSupportMock).when(bridge, "setup");
+        bridge.init(txMock, getGenesisBlock(), null, null, null, null);
+
+        byte[][] headers = new byte[][] { Hex.decode(
+                "0000002006226e46111a0b59caaf126043eb5bbf28c34f3a5e332a1fc7b2b73cf188910ff698ee112158f5573a90b7403cfba074addd61c547b3639c6afdcf52588eb8e2a1ef825cffff7f2000000000"
+        ) };
+
+        byte[] data = BridgeMethods.RECEIVE_HEADERS.getFunction().encode(new Object[]{ headers });
+
+        boolean failed = false;
+        try {
+            bridge.execute(data);
+        } catch (RuntimeException e) {
+            Assert.assertTrue(e.getMessage().contains("Sender is not part of the active"));
+            failed = true;
+        }
+        Assert.assertTrue(failed);
+        verify(bridgeSupportMock, never()).receiveHeaders(any(BtcBlock[].class));
+    }
+
+    @Test
+    public void receiveHeadersAccess_beforePublic_accessIfFromFederationMember() throws Exception {
+        doReturn(false).when(activationConfig).isActive(eq(RSKIP124), anyLong());
+
+        RskAddress sender = new RskAddress(ECKey.fromPrivate(HashUtil.keccak256("federator1".getBytes(StandardCharsets.UTF_8))).getAddress());
+        Transaction txMock = mock(Transaction.class);
+        when(txMock.getReceiveAddress()).thenReturn(PrecompiledContracts.BRIDGE_ADDR);
+        when(txMock.getSender()).thenReturn(sender);
+
+        BridgeSupport bridgeSupportMock = mock(BridgeSupport.class);
+        when(bridgeSupportMock.getRetiringFederation()).thenReturn(null);
+        when(bridgeSupportMock.getActiveFederation()).thenReturn(BridgeRegTestConstants.getInstance().getGenesisFederation());
+
+        Bridge bridge = PowerMockito.spy(new Bridge(PrecompiledContracts.BRIDGE_ADDR, constants, activationConfig));
+        PowerMockito.doReturn(bridgeSupportMock).when(bridge, "setup");
+        bridge.init(txMock, getGenesisBlock(), null, null, null, null);
```

Comment:
Maybe in another PR would be nice to have a function or class to abstract Bridge and BridgeSupport creation for tests

Reply:
agreed, we will be doing that after we wrap up this release